client/pseudonymsys_ca.go

```diff
@@ -30,28 +30,29 @@ import (
 
 type PseudonymsysCAClient struct {
 	genericClient
-	prover *dlogproofs.SchnorrProver
+	grpcClient pb.PseudonymSystemCAClient
+	prover     *dlogproofs.SchnorrProver
 }
 
 func NewPseudonymsysCAClient(conn *grpc.ClientConn) (*PseudonymsysCAClient, error) {
 	group := config.LoadGroup("pseudonymsys")
-	genericClient, err := newGenericClient(conn)
-	if err != nil {
-		return nil, err
-	}
 
 	return &PseudonymsysCAClient{
-		genericClient: *genericClient,
+		genericClient: newGenericClient(),
+		grpcClient:    pb.NewPseudonymSystemCAClient(conn),
 		prover:        dlogproofs.NewSchnorrProver(group, protocoltypes.Sigma),
 	}, nil
 }
 
 // ObtainCertificate provides a certificate from trusted CA to the user. Note that CA
```

Comment:
Correct comment to `GenerateCertificate`.
On the other hand, I'm not sure I like the new name. How come you decided to rename this method?

Reply:
Done, corrected comment here and in EC implementation as well.

As for the name, I actually modified server-side handlers (that were previously named `PseudonymsysCA` and `PseudonymsysCAEC`) before I got to the clients. So I noticed afterwards that on client-side they were indeed named `ObtainCertificate`. But I think the distinction is subtle, and since we have `GenerateNym`, and not `ObtainNym`, why not have `GenerateCertificate` as well? So I just went for consistency, although I have to admit I didn't put too much thought in it :slightly_smiling_face:

Reply:
Well, we don't have total consistency anyway. There's `GenerateNym`, `ObtainCredentials`, and `Obtain/GenerateCertificate`. It would seem logical to me that "obtain" implies getting something from the server, while "generate" means a local operation - although I know it's not like this.
And since names `GenerateNym` and `ObtainCredentials` are used in the protocol definitions, we should keep them. I guess it doesn't really matter for Certificates then :)